build/preview.ts

```diff
@@ -0,0 +1,40 @@
+import * as fs from 'fs';
+import * as path from 'path';
+
+async function run() {
+  const cwd = path.resolve(__dirname);
+
+  const buildDir = path.basename(cwd);
+  const repoDir = cwd.replace(buildDir, '');
+
+  // Patch README
+  const data = fs.readFileSync(path.join(repoDir, 'README.md'), { encoding: 'utf8' });
+  const insert = fs.readFileSync(path.join(buildDir, 'nightly.md'), { encoding: 'utf8' });
+  fs.writeFileSync(path.join(repoDir, 'README.md'), `${insert}\n${data}`);
+
+  // Patch package.json
+  const date = new Date(new Date().toLocaleString('en-US', { timeZone: 'America/New_York' }));
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  let json = require('../package.json');
+
+  const version = `${String(date.getFullYear())}.${date.getMonth() + 1}.${date.getDate()}${String(
```

Comment:
If we are using a date here instead of the date of the last git commit, we'll publish a new version every night, even if nothing was changed. Is this intended?

Reply:
Good point. So, we should we prevent the run from executing in Github instead of relying on git log behavior to prevent this.

Reply:
With the new matrix approach, we are able to still use the bash scripts